lib/aws-manager.js

```diff
@@ -1247,7 +1125,9 @@ class AwsManager {
 
     let perRegionKills = {};
 
-    for (let region of this.ec2.regions) {
+    let killPromises = [];
+
+    for (let region of _.keys(this.ec2)) {
```

Comment:
>
> I like the idea of SmartFilter, but I think we can get just as clear
> results with what I've changed to doing:

Yeah, it needs a better name than "SmartFilter"... maybe just "filter".
The idea with it was that we could use it for a bunch of things...

There are many places where you need to filter in the manner you do here...
Specially when looking at either results from ec2, or apiState...

I think I've said it before... but I would have merged all entries in
apiState properties into one array...
Even the different types... and then I would have .type = 'spot-request' or
.type = 'instance', etc...

That's a larger thing.. but I  suspect you could get a nice design.
Basically treating apiState = [...] as a nosql database, that you just
query..

Anyways, feel free to ignore it... but you have data that you look at in
different ways, a big hammer is probably useful :)

--
Regards Jonas Finnemann Jensen.

2016-02-23 3:29 GMT-08:00 John Ford <notifications@github.com>:

> In lib/aws-manager.js
> <https://github.com/taskcluster/aws-provisioner/pull/55#discussion_r53767882>
> :
>
> > @@ -1247,7 +1125,9 @@ class AwsManager {
> >
> >      let perRegionKills = {};
> >
> > -    for (let region of this.ec2.regions) {
> > +    let killPromises = [];
> > +
> > +    for (let region of _.keys(this.ec2)) {
>
> I like the idea of SmartFilter, but I think we can get just as clear
> results with what I've changed to doing:
>
>       let instances = this.__apiState.instances.filter(instance => {
>         return instance.WorkerType === name && states.includes(instance.State.Name);
>       });
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/taskcluster/aws-provisioner/pull/55/files#r53767882>.
>